plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,564 @@
+        }
+
+        @Override
+        public UpdateProperties updateProperties()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ReplaceSortOrder replaceSortOrder()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public UpdateLocation updateLocation()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public AppendFiles newAppend()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public RewriteFiles newRewrite()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public RewriteManifests rewriteManifests()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public OverwriteFiles newOverwrite()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public RowDelta newRowDelta()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ReplacePartitions newReplacePartitions()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public DeleteFiles newDelete()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ExpireSnapshots expireSnapshots()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Rollback rollback()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ManageSnapshots manageSnapshots()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Transaction newTransaction()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public FileIO io()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public EncryptionManager encryption()
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public LocationProvider locationProvider()
+        {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    private static class TestingSnapshot
+            implements Snapshot
+    {
+        @Override
+        public long sequenceNumber()
+        {
+            return 0;
+        }
+
+        @Override
+        public long snapshotId()
+        {
+            return 0;
+        }
+
+        @Override
+        public Long parentId()
+        {
+            return 0L;
+        }
+
+        @Override
+        public long timestampMillis()
+        {
+            return 0;
```

Comment:
`throw new UnsupportedOperationException();` ?

or is this actually needed

Reply:
Removed this class.

Reply:
I removed this class but I think it is still good to keep: #9050 (comment).